R/initialisers.R

```diff
@@ -0,0 +1,67 @@
+#' init_analytics
+#' @description
+#' Creates the google-analytics.html script in order to allow the activation of
+#' analytics via GA4. For the full steps required to set up analytics, please
+#' refer to the documentation in the readme.
+#'
+#' @param ga_code The Google Analytics code
+#' @importFrom magrittr %>%
+#' @return TRUE if written, FALSE if not
+#' @export
+#'
+#' @examples init_analytics(ga_code = "0123456789")
+init_analytics <- function(ga_code) {
+  is_valid_ga4_code <- function(ga_code) {
+    stringr::str_length(ga_code) == 10 & typeof(ga_code) == "character"
+  }
+
+  if (is_valid_ga4_code(ga_code) == FALSE) {
+    stop(
+      'You have entered an invalid GA4 code in the ga_code argument.
+      Please enter a 10 digit code as a character string.
+      e.g. "0123QWERTY"'
+    )
+  }
+
+  webpage <- RCurl::getURL("https://raw.githubusercontent.com/dfe-analytical-services/dfeshiny/analytsics-init/inst/google-analytics.hml")
```

Comment:
You have .hml here too, typo or genuine?

Reply:
Fixed

Reply:
I don't think you have? Can still see it on line 30, though I guess that line needs updating to point to the main branch too?